portal-api/src/v1/gef/controllers/files.controller.js

```diff
@@ -62,7 +62,7 @@ exports.create = async (req, res) => {
 
   // Check deal exists
   const dealCollection = await db.getCollection(dealCollectionName);
-  const deal = await dealCollection.findOne({ _id: ObjectId(String(parentId)) });
+  const deal = await dealCollection.findOne({ _id: { $eq: ObjectId(String(parentId)) } });
```

Comment:
Can we please add ObjectId validation and remove `String()` from all of the instances.